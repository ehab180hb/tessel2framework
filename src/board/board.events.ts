import * as events from 'events'
export const boardEvents = new events()

export enum BoardEvents {
  UPDATE_BUTTONS = 'updateButtons',
  UPDATE_SCREEN = 'updateScreen',
}
