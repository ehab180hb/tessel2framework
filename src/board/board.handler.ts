import * as five from 'johnny-five'

const isTesting = process.env.NODE_ENV === 'test'

const Tessel = !isTesting ? require('tessel-io') : class {}

export const getBoard = () => {
  return new five.Board({
    io: new Tessel(),
  })
}
