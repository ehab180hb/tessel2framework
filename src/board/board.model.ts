export enum PIN {
  'a0',
  'a1',
  'a2',
  'a3',
  'a4',
  'a5',
  'a6',
  'a7',
  'b0',
  'b1',
  'b2',
  'b3',
  'b4',
  'b5',
  'b6',
  'b7',
}
