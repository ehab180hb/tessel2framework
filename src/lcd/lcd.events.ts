import { newMessage } from '../messages/messages.handler'
import { boardEvents, BoardEvents } from '../board/board.events'
import { fiveTypes } from '../typings/johnny-five'

interface UpdateScreenInput {
  row1?: string
  row2?: string
  lcd: fiveTypes.LCD
}

export const attachScreenUpdate = (updateScreenParams: UpdateScreenInput): void => {
  const emptyRow = ' '.repeat(16)
  const { row1 = emptyRow, row2 = emptyRow, lcd } = updateScreenParams
  const writeMsg = newMessage(lcd)
  writeMsg(0, row1)
  writeMsg(1, row2)
}

export const updateScreen = (updateScreenParams: UpdateScreenInput) => {
  boardEvents.emit(BoardEvents.UPDATE_SCREEN, updateScreenParams)
}
