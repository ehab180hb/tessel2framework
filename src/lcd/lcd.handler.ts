import * as five from 'johnny-five'
import { PIN } from '../board/board.model'
import { fiveTypes } from '../typings/johnny-five'

const pins: PIN[] = [PIN.a2, PIN.a3, PIN.a4, PIN.a5, PIN.a6, PIN.a7]

export const getLcd = (): fiveTypes.LCD => new five.LCD({ pins }) as any
