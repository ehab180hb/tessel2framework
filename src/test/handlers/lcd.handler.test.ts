import * as five from 'johnny-five'
import { PIN } from '../../board/board.model'
import { getLcd } from '../../lcd/lcd.handler'

const pins: PIN[] = [PIN.a2, PIN.a3, PIN.a4, PIN.a5, PIN.a6, PIN.a7]

describe('LCD handler', () => {
  beforeEach(() => {
    ;(five.LCD as any) = jest.fn()
  })

  describe('getBoard()', () => {
    test('returns properly instantiated LCD', () => {
      const lcd = getLcd()
      expect(lcd).toMatchSnapshot()
      expect(five.LCD).toHaveBeenCalledWith({
        pins,
      })
    })
  })
})
