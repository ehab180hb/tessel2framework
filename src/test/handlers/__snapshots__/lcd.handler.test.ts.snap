// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LCD handler getBoard() returns properly instantiated LCD 1`] = `mockConstructor {}`;
